lib/ansible/constants.py

```diff
@@ -56,8 +51,31 @@ def __getitem__(self, y):
         return self._value[y]
 
 
+def __getattr__(config_constant):
+    ''' Handle dynamicall generating a 'constant' when first requested,
+        otherwise just return it from cached value.
+    '''
+
+    if config_constant not in globals():
+        try:
+            value = config.get_config_value(config_constant, variables=globals())
+        except Exception as e:
+            raise AttributeError(e)
+
+        globals()[config_constant] = value
+
+        if config.WARNINGS:
+            for warn in config.WARNINGS.pop():
+                _warning(warn)
```

Comment:
Previously, this would run once. But now, it re-runs for every var. Also, why do you iterate through the popped warning value instead of the warnings list?
Did you mean to do something like
```suggestion
        if config.WARNINGS and (warn := config.WARNINGS.pop()):
            _warning(warn)
```
?

Reply:
more than one warning can occur, so realy should be:

```
while warn = config.warnings.pop():
```